homework/src/index.js

```diff
@@ -19,29 +20,128 @@
   function createAndAppend(name, parent, options = {}) {
     const elem = document.createElement(name);
     parent.appendChild(elem);
-    Object.keys(options).forEach((key) => {
-      const value = options[key];
-      if (key === 'text') {
-        elem.innerText = value;
+    if (options != null) {
+      Object.keys(options).forEach((key) => {
+        const value = options[key];
+        if (key === 'text') {
+          elem.innerText = value;
+        } else {
+          elem.setAttribute(key, value);
+        }
+      });
+    }
+    return elem;
+  }
+
+  function sortData(data, key) {
+    const sortedData = data.sort(function (a, b) {
+      let elem1 = a[key].toLowerCase();
+      let elem2 = b[key].toLowerCase();
+      if (elem1 < elem2) { return -1; }
+      if (elem1 > elem2) { return 1; }
+      return 0;
+    });
+    return sortedData;
+  }
+
+  function fillRepositoryList(data, root) {
+    const select = createAndAppend('select', root, { id: 'selectRepositories' });
+    const sortedData = sortData(data, 'name');
+    sortedData.forEach(elem => {
+      createAndAppend('option', select, { text: elem.name, value: elem.id });
+    });
+
+    //Listener For Repository
+    document.getElementById('selectRepositories').addEventListener("change", e => {
+      if (divRepDetails.hasChildNodes) {
+        divRepDetails.remove();
+      }
+      detailedInfoRepository();
+    });
+
+    return sortedData;
+  }
+
+  function fillUser(dataUser, root) {
+    //const sortedUser = sortData(dataUser, 'login');
+    dataUser.forEach(user => {//sortedUser
+
+      fetchJSON(user.url, (err, user1) => {
```

Comment:
Hi @Damla78, this is not yet a full review of your homework, but just a comment on why you are hitting the API limit resulting in a 403 error. 

In this `forEach()` method you are making an additional request for each contributor. For instance, for the JavaScript1 repository there are 21 contributors. Therefore, for this repository you will make 21 + 1 (for the `contributors_url`) requests. Do this three times and you will hit the API limit.

However, it is not necessary to do a request per contributor because all the information you need is already in the response that you get from the `contributors_url` query. It includes the name, number of contributors and a link to the avatar picture. No need for an extra request per contributor.

Reply:
Thank you Jim. I got it.